MapboxCoreNavigation/RouteController.swift

```diff
@@ -85,7 +84,7 @@ open class RouteController: NSObject, Router {
             return routeProgress.route
         }
         set {
-            routeProgress = RouteProgress(route: newValue)
+            routeProgress = RouteProgress(route: newValue, previousRouteProgress: shouldDiscardHistory ? nil : routeProgress)
```

Comment:
Should we be storing the entire previous route progress object, or just a skeletal one that includes only the information we care about? Can we get away with setting `previousRouteProgress` to a RouteProgress that’s been flattened into having a single leg with a single step, to reduce memory usage?